cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -251,266 +229,35 @@ While all the techniques referenced here do not require any user interaction, so
-                " is absent/empty so we block the request !";
-                LOG.warn(accessDeniedReason);
-                httpResp.sendError(HttpServletResponse.SC_FORBIDDEN, accessDeniedReason);
-            } else if (!tokenFromHeader.equals(tokenCookie.getValue())) {
-                //Verify that token from header and one from cookie are the same
-                //Here is not the case so we trace the event and we block the request
-                accessDeniedReason = "CSRFValidationFilter: Token provided via HTTP Header"+ 
-                "and via Cookie are not equals so we block the request !";
-                LOG.warn(accessDeniedReason);
-                httpResp.sendError(HttpServletResponse.SC_FORBIDDEN, accessDeniedReason);
-            } else {
-                //Verify that token from header and one from cookie matches
-                //Here is the case so we let the request reach the target component 
-                //(ServiceServlet, jsp...) and add a new token when we get back the bucket
-                HttpServletResponseWrapper httpRespWrapper = 
-                                            new HttpServletResponseWrapper(httpResp);
-                chain.doFilter(request, httpRespWrapper);
-                //Add the CSRF token cookie and header
-                this.addTokenCookieAndHeader(httpReq, httpRespWrapper);
-            }
-        }
-    }
+Login CSRF can be mitigated by creating pre-sessions (sessions before a user is authenticated) and including tokens in login form. You can use any of the techniques mentioned above to generate tokens. Pre-sessions can be transitioned to real sessions once the user is authenticated. This technique is described in [Robust Defenses for Cross-Site Request Forgery section 4.1](https://seclab.stanford.edu/websec/csrf/csrf.pdf).
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void init(FilterConfig filterConfig) throws ServletException {
-        //To easier the configuration, we load the target expected origin from 
-        //an JVM property
-        //Reconfiguration only require an application restart that is generally acceptable
-        try {
-            this.targetOrigin = new URL(System.getProperty(TARGET_ORIGIN_JVM_PARAM_NAME));
-        } catch (MalformedURLException e) {
-            LOG.error("Cannot init the filter !", e);
-            throw new ServletException(e);
-        }
-        LOG.info("CSRFValidationFilter: Filter init, set expected target origin to '{}'.", 
-                 this.targetOrigin);
-    }
+If sub-domains under your master domain are treated as not trusty in your threat model, it is difficult to mitigate login CSRF. A strict subdomain and path level referrer header (because most login pages are served on HTTPS - no stripping of referrer - and are also linked from home pages) validation (detailed in section 6.1) can be used in these cases for mitigating CSRF on login forms to an extent.
```

Comment:
`not trusted`

Not sure what you're saying in the `(because most login pages....)` section

What's section 6.1?